experimenter/experimenter/jetstream/models.py

```diff
@@ -159,34 +159,39 @@ def append_retention_data(self, weekly_data):
 
 
 class DataPoint(BaseModel):
-    lower: float = None
-    upper: float = None
-    point: float = None
-    window_index: str = None
-    count: float = None
+    lower: float | None = None
+    upper: float | None = None
+    point: float | None = None
+    window_index: str | None = None
+    count: float | None = None
 
-    def set_window_index(self, window_index):
-        self.window_index = window_index
+    @field_validator("window_index", mode="before")
+    @classmethod
+    def coerce_window_index(cls, v: Any):
+        if isinstance(v, int):
+            return str(v)
+
+        return v
 
     def has_bounds(self) -> bool:
         return self.lower and self.upper
 
 
 class BranchComparisonData(BaseModel):
-    all: list[DataPoint] = []
-    first: DataPoint = DataPoint()
+    all: list[DataPoint] = Field(default_factory=list)
+    first: DataPoint = Field(default_factory=DataPoint)
 
 
 class SignificanceData(BaseModel):
-    overall: dict[str, Any] = {}
-    weekly: dict[str, Any] = {}
+    overall: dict[str, Any] = Field(default_factory=dict)
+    weekly: dict[str, Any] = Field(default_factory=dict)
 
 
 class MetricData(BaseModel):
-    absolute: BranchComparisonData
-    difference: BranchComparisonData
-    relative_uplift: BranchComparisonData
-    significance: SignificanceData
+    absolute: BranchComparisonData = Field(default_factory=BranchComparisonData)
+    difference: BranchComparisonData = Field(default_factory=BranchComparisonData)
+    relative_uplift: BranchComparisonData = Field(default_factory=BranchComparisonData)
+    significance: SignificanceData = Field(default_factory=BranchComparisonData)
```

Comment:
Looking through this PR to find out why there was an error when deployed and found this bug:

The default for `significance` should be a `SignificanceData` not a `BranchComparisonData`.